### DISCUSSION TITLE
About implementing all SSH calls to different client/server architecture.

### DISCUSSION THREAD

**Discussion by h-alice** [NatLee/telepy#18] [Q&A] [answered]
@NatLee Since I want to use a different SSH client/server implementation (maybe will be in GO).
Can you list all the direct call of SSH (such as `ssh some_server -l some_user`)?

I'll post some test result if I make some progress.

**Comment by NatLee** [comment, marked as answer]
The endpoint exclusively utilises SSH or AutoSSH, which is essentially the same as SSH, to connect to our Telepy SSH server.


Endpoint --- Tunnel (SSH) forwards to an empty port in SSH server ---> Telepy SSH server


When using the Telepy Web server in the Web terminal, the Telepy SSH server acts as a proxy server and connects to the endpoint server.


Web Terminal -- SSH Proxy --> Telepy SSH server -- SSH connect to localhost --> Endpoint


As the user is on the endpoint, we cannot simply list them.

In my opinion, by implementing the endpoint tool, the system can just use local user and notify the Telepy Web server to retrieve it.🤩

**Comment by h-alice** [comment]
Thanks for the reply, this answered my question!
